### DISCUSSION TITLE
Is there a noticeable difference?

### DISCUSSION THREAD

**Discussion by tchleb** [awawa-dev/HyperSPI#1] [General]
Is there are a visible difference when I use hyperSPI to control my sk6812 Leds?

My setup is following:

-Raspberry pi zero W
-USB video grabber
-220x sk6812 rgbnw are connected with shielded cable to raspberry pi gpio (spi) 
-Baudrate is 3 Megabits

br
Tobias

**Comment by awawa-dev** [comment]
Hi
1) Do you have a problem with a USB grabber bandwidth (usbmon)? 
2) You use direct access (although not supported anymore) so every external solution introduce some lag. With HyperSPI it's minimal because of using 25Mb speed.
3) Most import because you are using Rpi zero: resources. Using rpi_ws821x consumes resources for preparing a buffer for emulating neopixel protocol. How much does it cost? Quite a lot in fact. Even with the incoming patch [link](https://github.com/jgarff/rpi_ws281x/pull/468) (LUT table btw 😉 ) it's still a lot. So it's better to delegate that task for external device.

**Reply by tchleb** [reply]
Thank you for your reply.
Yesterday I tried HyperSPI, I would say the performance is significant better :) 

![IMG_20210923_213055__01.jpg](https://user-images.githubusercontent.com/45948590/134649514-1d1614ca-9d5f-462f-a071-b9a5d2183a2b.jpg)
